modules/auxiliary/cloud/aws/s3_enumerate.rb

```diff
@@ -0,0 +1,110 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'aws-sdk-s3'
+
+class MetasploitModule < Msf::Auxiliary
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'        => 'Amazon Web Services S3 instance enumeration',
+        'Description' => %q(
+                          Provided AWS credentials, this module will call the authenticated
+                          API of Amazon Web Services to list all S3 buckets associated
+                          with the account
+                         ),
+        'Author'      => ['Aaron Soto <aaron.soto@rapid7.com>'],
+        'License'     => MSF_LICENSE
+      )
+    )
+
+    register_options(
+      [
+        OptInt.new('LIMIT', [false, 'Only return the specified number of results']),
+        OptString.new('REGION', [false, 'AWS Region (eg. "us-west-2")']),
+        OptString.new('ACCESS_KEY_ID', [true, 'AWS Access Key ID (eg. "AKIAXXXXXXXXXXXXXXXX")', '']),
+        OptString.new('SECRET_ACCESS_KEY', [true, 'AWS Secret Access Key (eg. "CA1+XXXXXXXXXXXXXXXXXXXXXX6aYDHHCBuLuV79")', ''])
+      ]
+    )
+  end
+
+  def handle_aws_errors(e)
+    if e.class.parents.include?(Aws)
+      fail_with(Failure::UnexpectedReply, e.message)
+    else
+      raise e
+    end
+  end
+
+  def describe_s3_bucket(i)
+    print_good "  Name:           #{i.name}"
+    print_good "  Creation Date:  #{i.creation_date}"
+    print_good "  # of Objects:   #{@s3.list_objects_v2(bucket: i.name).contents.length}"
+    print_good "  Region:         #{@s3.get_bucket_location(bucket: i.name).location_constraint}"
+
+    begin
+      print_good "  Website:        /#{@s3.get_bucket_website(bucket: i.name).index_document.suffix}"
+    rescue Aws::S3::Errors::NoSuchWebsiteConfiguration
+      print_good "  Website:        (None)"
+    end
+
+    acl = @s3.get_bucket_acl(bucket: i.name)
+    print_good "  Owner:          #{acl.owner.display_name}"
+    print_good "  Permissions:"
+    acl.grants.each do |i|
+
+      if i.grantee.type == "CanonicalUser"
```

Comment:
Why change CanonicalUser to User ?

Reply:
No reason.  It just read more easily.  😄